src/core/Akka.Persistence/AtLeastOnceDeliveryReceiveActor.cs

```diff
@@ -52,31 +55,40 @@ protected AtLeastOnceDeliveryReceiveActor(PersistenceSettings.AtLeastOnceDeliver
         /// this helps prevent an overwhelming amount of messages to be sent at once.
         /// 
         /// The default value can be configure with the 'akka.persistence.at-least-once-delivery.redelivery-burst-limit'
-        /// configuration key. This method can be overridden by implementation classes to return
-        /// non-default values.
+        /// configuration key.
         /// </summary>
-        public int RedeliveryBurstLimit => _atLeastOnceDeliverySemantic.RedeliveryBurstLimit;
+        public int RedeliveryBurstLimit
+        {
+            get => _atLeastOnceDeliverySemantic.RedeliveryBurstLimit;
+            set => _atLeastOnceDeliverySemantic.RedeliveryBurstLimit = value;
+        }
 
         /// <summary>
         /// After this number of delivery attempts a <see cref="UnconfirmedWarning" /> message will be sent to
         /// <see cref="ActorBase.Self" />. The count is reset after restart.
         /// 
         /// The default value can be configure with the 'akka.persistence.at-least-once-delivery.warn-after-number-of-unconfirmed-attempts'
-        /// configuration key. This method can be overridden by implementation classes to return
-        /// non-default values.
+        /// configuration key.
         /// </summary>
-        public int WarnAfterNumberOfUnconfirmedAttempts => _atLeastOnceDeliverySemantic.WarnAfterNumberOfUnconfirmedAttempts;
+        public int WarnAfterNumberOfUnconfirmedAttempts
+        {
+            get => _atLeastOnceDeliverySemantic.WarnAfterNumberOfUnconfirmedAttempts;
+            set => _atLeastOnceDeliverySemantic.WarnAfterNumberOfUnconfirmedAttempts = value;
+        }
 
         /// <summary>
         /// Maximum number of unconfirmed messages, that this actor is allowed to hold in the memory.
         /// if this number is exceeded, <see cref="AtLeastOnceDeliverySemantic.Deliver" /> will not accept more
         /// messages and it will throw <see cref="MaxUnconfirmedMessagesExceededException" />.
         /// 
         /// The default value can be configure with the 'akka.persistence.at-least-once-delivery.max-unconfirmed-messages'
-        /// configuration key. This method can be overridden by implementation classes to return
-        /// non-default values.
+        /// configuration key.
         /// </summary>
-        public int MaxUnconfirmedMessages => _atLeastOnceDeliverySemantic.MaxUnconfirmedMessages;
+        public int MaxUnconfirmedMessages
+        {
+            get => _atLeastOnceDeliverySemantic.MaxUnconfirmedMessages;
+            set => _atLeastOnceDeliverySemantic.MaxUnconfirmedMessages = value;
```

Comment:
All of these settings should be immutable and overloadable via the CTOR, as you mentioned @ondrejpialek